### PR TITLE
Social Login: Cleanup Facebook

### PR DESCRIPTION
**Pull request by samisuteria**
- This adds Firebase Login and implements Facebook login
- Another PR is coming for Google and Apple (it was getting really messy)

### CODE REVIEW

File: Steps4Impact/App/AppController.swift [hunk 1]
```diff
@@ -69,8 +69,8 @@ class AppController {
 
     // Select Default View
     // TODO(samisuteria) add check for akf profile?
-    print("Facebook Id: \(Facebook.id)")
-    if Facebook.id.isEmpty {
+    let currentUser = Auth.auth().currentUser
```
Comment (compnerd) [inline Steps4Impact/App/AppController.swift]: Is this copy to avoid some sort of reference count issue?  Seems like its a single use variable, why not inline it?

File: Steps4Impact/LoginV2/LoginV2ViewController.swift [hunk 2]
```diff
@@ -0,0 +1,216 @@
+/**
+* Copyright © 2019 Aga Khan Foundation
```
Comment (compnerd) [inline Steps4Impact/LoginV2/LoginV2ViewController.swift]:
```suggestion
* Copyright © 2020 Aga Khan Foundation
```

File: Steps4Impact/LoginV2/LoginV2ViewController.swift [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+      onMain {
+        AppController.shared.login()
+      }
+      self?.addParticipantToDefaultEventIfNeeded(fbid: fbid)
+    }
+  }
+  
+  private func addParticipantToDefaultEventIfNeeded(fbid: String) {
+    AKFCausesService.getParticipant(fbid: fbid) { (result) in
+      if let participant = Participant(json: result.response), participant.currentEvent == nil {
```
Comment (compnerd) [inline Steps4Impact/LoginV2/LoginV2ViewController.swift]:
```suggestion
      guard let participant = Participant(json: result.response) else { return }
      guard participant.currentEvent == nil else { return }
```

Comment (compnerd) [reply Steps4Impact/LoginV2/LoginV2ViewController.swift]: Using guards a bit more heavily would allow you to remove the indentation.

File: Steps4Impact/LoginV2/LoginV2ViewController.swift [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+          self.linkAKFBackend(fbid: result.user.uid)
+        } else {
+          // This shouldn't happen and we don't have a way to recover except trying again.
+          self.showFirebaseLoginErrorAlert(body: "Unknown Error")
+        }
+        
+        return
+      }
+      
+      // Don't handle errors
```
Comment (compnerd) [inline Steps4Impact/LoginV2/LoginV2ViewController.swift]: This seems confusing, can you rephrase it?  The error has been "handled" above hasn't it?

File: Steps4Impact/LoginV2/LoginV2ViewController.swift [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+  @objc
+  private func facebookLoginButtonTapped() {
+    let loginManager = LoginManager()
+    loginManager.logIn(permissions: [.publicProfile], viewController: self) { [weak self] (result) in
+      guard let `self` = self else { return }
+      switch result {
+      case .success(granted: _, declined: _, token: let accessToken):
+        let credential = FacebookAuthProvider.credential(withAccessToken: accessToken.tokenString)
+        self.linkFirebase(credential)
+        print(credential)
```
Comment (compnerd) [inline Steps4Impact/LoginV2/LoginV2ViewController.swift]: Debugging left overs?

File: Steps4Impact/App/User.swift [hunk 6]
```diff
@@ -0,0 +1,35 @@
+/**
+* Copyright © 2019 Aga Khan Foundation
```
Comment (compnerd) [inline Steps4Impact/App/User.swift]:
```suggestion
* Copyright © 2020 Aga Khan Foundation
```

File: Steps4Impact.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -27,7 +27,6 @@
 		260BF5B923250414005F0882 /* SettingsTitleCell.swift in Sources */ = {isa = PBXBuildFile; fileRef = 260BF5B823250414005F0882 /* SettingsTitleCell.swift */; };
 		2611A4EF2335F8350030E77E /* Steps4ImpactSpec.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2611A4EE2335F8350030E77E /* Steps4ImpactSpec.swift */; };
 		2611A5052335FF290030E77E /* AKFCausesService.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2611A5012335FF290030E77E /* AKFCausesService.swift */; };
-		2611A5062335FF290030E77E /* Facebook.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2611A5022335FF290030E77E /* Facebook.swift */; };
```
Comment (compnerd) [inline Steps4Impact.xcodeproj/project.pbxproj:30]: Shouldn't this happen after the references are removed?  This commit wont build.

File: Steps4Impact/Networking/AKFCausesService.swift [hunk 8]
```diff
@@ -127,6 +130,11 @@ class AKFCausesService: Service {
                                 completion: ServiceRequestCompletion? = nil) {
     shared.request(.delete, endpoint: .participant(fbId: fbid), completion: completion)
   }
+  
+  static func getParticipantSocial(fbid: String,
+                                   completion: ServiceRequestCompletion? = nil) {
+    shared.request(endpoint: .participantSocial(fbId: fbid), completion: completion)
+  }
```
Comment (compnerd) [inline Steps4Impact/Networking/AKFCausesService.swift:137]: Is this still needed after the firebase switch?

File: Steps4Impact/App/AppController.swift (hunk 1 above)
Comment (samisuteria) [reply Steps4Impact/App/AppController.swift]: Good call - I'll update this to use the new `User.id` helper instead

File: Steps4Impact/LoginV2/LoginV2ViewController.swift (hunk 4 above)
Comment (samisuteria) [reply Steps4Impact/LoginV2/LoginV2ViewController.swift]: the guard above is actually backwards above. I'll refactor it

File: Steps4Impact/LoginV2/LoginV2ViewController.swift (hunk 3 above)
Comment (samisuteria) [reply Steps4Impact/LoginV2/LoginV2ViewController.swift]: good call

File: Steps4Impact/LoginV2/LoginV2ViewController.swift (hunk 5 above)
Comment (samisuteria) [reply Steps4Impact/LoginV2/LoginV2ViewController.swift]: yep - removing

File: Steps4Impact.xcodeproj/project.pbxproj (hunk 7 above)
Comment (samisuteria) [reply Steps4Impact.xcodeproj/project.pbxproj:30]: good call - I'll reorder the commits

File: Steps4Impact/Networking/AKFCausesService.swift (hunk 8 above)
Comment (samisuteria) [reply Steps4Impact/Networking/AKFCausesService.swift:137]: yep - only the current user's displayName and photoURL can be seen by the current user. Team Member/Leaderboard images are loaded using firebase's auth sdk from our api. Matching PR for that is here: https://github.com/AgaKhanFoundation/WCB/pull/148

File: Steps4Impact.xcodeproj/project.pbxproj (hunk 7 above)
Comment (samisuteria) [reply Steps4Impact.xcodeproj/project.pbxproj:30]: not sure why GitHub isn't reflecting the new order on the PR but I moved the `delete Facebook services` commit on top of the changes. 
Screen Shot 2020-06-28 at 6 25 22 PM

File: Steps4Impact/App/User.swift (hunk 6 above)
Comment (samisuteria) [reply Steps4Impact/App/User.swift]: updated

**Review by aalimmulji — APPROVED** [review]
